### DISCUSSION TITLE
Anywidgets or Jupyter Widgets

### DISCUSSION THREAD

**Discussion by yilinxia** [arviz-devs/arviz_dashboard#26] [Ideas]
Hey everyone,

I've been pondering some long-term considerations, especially as we navigate our dashboard widget integration. We have two choices on the table: `anywidgets` and `jupyter widgets`. From what I've observed, `jupyter widgets` are quite popular in packages like `Jupyter Voila` and `Panel`, offering a solid range of features. However, I'm wondering about their adaptability to our unique requirements for the ArviZ dashboard.

Has anyone had experience with these or insights to share?

**Comment by aloctavodia** [comment]
I only have experience with ipywidgets and most of my experience is from very simple examples using interact for teaching or talks or from PreliZ, for instance, https://preliz.readthedocs.io/en/latest/examples/observed_space_examples.html or https://preliz.readthedocs.io/en/latest/examples/param_space_1d_examples.html#the-roulette-method you can see the examples are relatively simple and the layout/aesthetics still needs more work.

**Comment by ndmlny-qs** [comment]
anywidgets makes creating custom ipywidgets (jupyter widgets) a snap, since you have access to creating an ESM using anywidgets, and its build system. I'm a fan of the framework, and would like to explore it more. Perhaps making examples here is the next step, showing what one would do with panel/widgets/anywidget.

**Comment by manzt** [comment]
Would be happy to answer any questions about anywidget. If you support regular Jupyter Widgets, you should get anywidgets for free. It's worth mentioning that [Marimo added support for anywidget](https://twitter.com/marimo_io/status/1764762799467810906) only due to the more minimal API needed in the frontend.

**Comment by ndmlny-qs** [comment]
Thanks @manzt I would appreciate help getting up to speed with `anywidget`. I looked at the project again last week and think that a good place for me to start is getting the vite setup working.

One of the goals of the ArviZ dashboard project is to make it easy for a Bayesian modeler to visualize next steps they can take in their workflow. These visuals can also be used to showcase work done to stakeholders that are not experts with the Bayesian workflow. That means (to me at least) the visuals need to be easily rendered in a static page, which is doable, but definitely needs the build process of anywidget.

I have come to the conclusion (but still need to speak with the other members) that anywidget is the right tool to choose for this project. Panel is great, but if we want static rendering then we need to be able to run statistical computations in a browser, and not in a Python backend.
